src/store/slices/auth.ts

```diff
@@ -0,0 +1,74 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+import API from 'api';
+import { LoginFormType } from 'pages/Login/types';
+import { RegisterFormType } from 'pages/Register/types';
+
+export enum Statuses {
+  OK = 'OK',
+  ERROR = 'ERROR',
+}
+
+export const loginUser = createAsyncThunk(
+  'auth/loginUser',
+  async (user: LoginFormType) => {
+    try {
+      await API.createRequest('login', user);
+    } catch (error) {
+      throw new Error(error);
+    }
+  },
+);
+
+export const registerUser = createAsyncThunk(
+  'auth/registerUser',
+  async (user: RegisterFormType) => {
+    const { firstName, secondName, login, email, password, phone } = user;
+
+    try {
+      await API.createRequest('register', {
+        first_name: firstName,
+        second_name: secondName,
+        login,
+        email,
+        password,
+        phone,
+      });
+    } catch (error) {
+      throw new Error(error);
+    }
+  },
+);
+
+const slice = createSlice({
+  name: 'auth',
+  initialState: { loading: false, status: null },

```

Comment:
❗️
`initialState` может раздуваться, поэтому его имеет смысл определить выше, а тут присвоить
```suggestion
  const initialState =  { loading: false, status: null }
  // ...
  {
    initialState,

```